sfcasts/ep1/github-service-part-1.md

```diff
@@ -1,62 +1,69 @@
-yes, instead of two failures, we only have one. Our GitHub service test test GitHub
-test get health report returns, correct health status for Dino with data set sick
-Dino, it failed to asserting that two variables reference the same object. And it's
-of course doing that because we are only returning a healthy status for get health
-reports. Now that we have this kind of wired up, let's go ahead and implement the
-service.
+# GitHub Service - Part 1
 
+Now that we can see if a `Dinosaur` is accepting visitors on our dashboard, we
+need to keep the dashboard updated in real time by using the health status labels
+that GenLab has applied to several dino issues on GitHub. To do that we'll create 
+a service that will grab those labels from the issues using GitHub's API.
+
+## Test for our service first
+
+To test our new service, inside of `tests/Unit/` create a new `Service/` directory 
+and then a new class: `GithubServiceTest` which will extend `TestCase`. Add method
+`testGetHealthReportReturnsCorrectHealthStatusForDino` and inside, even
+though it doesn't exist yet, `$service = new GithubService()`.
+
+Our service will return a `HealthStatus` enum that is created from the health status
+label on GitHub, so we'll `assertSame()` that `$expectedStatus` is identical to 
+`$service->getHealthReport()`. Yes, we'll be using a data provider for this test...
+And because our service will return a *single* `HealthStatus` object, let's tell 
+the method which dino we need a health report for by passing in a `$dinoName` to
+`getHealthReport()`.
+
+Create a new method: `dinoNameProvider()` that returns a `\Generator`. Our first 
+dataset for the provider will have the key `Sick Dino`, which returns an array 
+with `HealthStatus::SICK` and `Daisy` for the dino's name. Next up is a 
+`Healthy Dino` with `HealthStatus::HEALTHY` who happens to be the one and only
+`Maverick`. On the test method, add a `@dataProvider` annotation so the test uses 
+the `dinoNameProvider` and we'll also pass in a `HealthStatus $expectedStatus`
+argument and a `string` called `$dinoName`.
+
+Move to your terminal and let's check out our tests:
+
+```terminal
+./vendor/bin/phpunit
+```
+
+And... Yup! Just as I expected, we have two errors because: 
+
+> The GithubService class cannot be found
+
+## Create the service that will call GitHub
+
+To fix those errors, inside of `src/`, create a new `Service/` directory. Then 
+we'll need a new class: `GithubService` and inside, add a method: `getHealthReport()`
+which takes a `string $dinosaurName` and gives back a `HealthStatus` object.
+
+Hmm... Alrighty, the first thing we will do is call GitHub's API to get the list
+of issues for the `dino-park` repository. Then we'll filter those issues to pick the 
+one that matches `$dinosaurName`. And last but not least, return 
+`GithubStatus::HEALTHY`, unless we find an issue with a `Status: Sick` label that
+matches our dino's name.
```

Comment:
It IS nice occasionally to give the user a game plan before we blindly code forward. I like that here. I rephrased it to literally start with "Here's the plan". I like that phrasing because it's like we're all huddling together on a team and making a clear plan before we execute.

Reply:
Edit: I see we don't code this until the next chapter (your comment below), so I've removed this paragraph entirely. For reference, it was:

> Here's the plan: we'll call GitHub's API to get the list of issues for the `dino-park`
> repository. Then we'll filter those issues to pick the  one that matches `$dinosaurName`.
> Finally, we'll return `GithubStatus::HEALTHY`, unless the issue has a `Status: Sick`
> label.

Reply:
NEVERMIND. I see what you do on the video now - and it's all good. Please put this paragraph back - I think it fits well.